src/index.js

```diff
@@ -111,8 +112,57 @@ import { Environment } from './environment';
     cssReapplier.reapply(['radio']);
   });
 
-  mutationObserver.observe(
-    document.querySelector(config.form.selector),
-    { childList: true, subtree: true },
-  );
+  mutationObserver.observe(document.querySelector(config.form.selector), {
+    childList: true,
+    subtree: true,
+  });
 })();
+
+let timer = 0;
+window.repopulateIngridientDataGrid = () => {
+  let _l = true;
+  const data = JSON.parse(localStorage.getItem('Label Buster'));
+  const IngredientData = JSON.parse(data.IngredientData);
+  const _isLoop = sessionStorage.getItem('IsLoopedOnce');
+  if (_isLoop == 'true') {
+    _l = false;
+  }
+  if (IngredientData && IngredientData.length > 0) {
+    timer = setTimeout(() => {
+      let ActiveTab = document.querySelector(
+        'button.qg-btn.btn-link.active.visited',
+      );
+      ActiveTab = JSON.parse(ActiveTab.getAttribute('data-detail'));
+      if (ActiveTab.page == 8) {
+        if (_l) {
+          sessionStorage.setItem('IsLoopedOnce', true);
+          IngredientData.forEach((e, i) => {
+            if (e.ingredient) {
+              document
+                .querySelector("button[ref='datagrid-IngredientData-addRow']")
+                .click();
```

Comment:
hmm, so let me get this straight, it was easier to user the front door to re'add the data instead of working out how the submission user return/load service worked..

Reply:
Changed the implementation. You can check the commit 48c714d